src/common/selftest/selftest_firstlayer.cpp

```diff
@@ -162,7 +162,9 @@ LoopResult CSelftestPart_FirstLayer::stateFilamentLoadWaitFinished() {
         // IPartHandler::SetFsmPhase(PhasesSelftest::);
         return LoopResult::RunNext;
     } else {
-        return LoopResult::GoToMark;
+        state_selected_by_user = StateSelectedByUser::Calib;
```

Comment:
In case of abort we  want to return  LoopResult::GoToMark.
You should use switch.